File: jdroid-gradle-commons-plugin/src/main/groovy/com/jdroid/gradle/commons/tasks/AbstractGitHubTask.java
package com.jdroid.gradle.commons.tasks;

import com.jdroid.github.IRepositoryIdProvider;
import com.jdroid.github.RepositoryId;
import com.jdroid.github.client.GitHubClient;

public abstract class AbstractGitHubTask extends AbstractTask {

	protected GitHubClient createGitHubClient() {
		GitHubClient client = new GitHubClient();
		client.setSerializeNulls(false);
		client.setOAuth2Token(getGitHubWriteToken());
		return client;
	}

	protected IRepositoryIdProvider getIRepositoryIdProvider() {
		return RepositoryId.create(getGitHubRepositoryOwner(), getGitHubRepositoryName());
	}

	protected String getGitHubWriteToken() {
		return getExtension().getGitHubWriteToken();
	}

	protected String getGitHubReadToken() {
		return getExtension().getGitHubReadToken();
	}

	protected String getGitHubRepositoryOwner() {
		return getExtension().getGitHubRepositoryOwner();
	}

	protected String getGitHubRepositoryName() {
		return getExtension().getGitHubRepositoryName();
	}

}

File: jdroid-gradle-commons-plugin/src/main/groovy/com/jdroid/gradle/commons/tasks/AbstractGitHubTask.kt
package com.jdroid.gradle.commons.tasks

import com.jdroid.github.IRepositoryIdProvider
import com.jdroid.github.RepositoryId
import com.jdroid.github.client.GitHubClient

abstract class AbstractGitHubTask : AbstractTask() {

    val iRepositoryIdProvider: IRepositoryIdProvider
        get() = RepositoryId.create(gitHubRepositoryOwner, gitHubRepositoryName)

    val gitHubWriteToken: String
        get() = extension.gitHubWriteToken

    val gitHubReadToken: String
        get() = extension.gitHubReadToken

    val gitHubRepositoryOwner: String
        get() = extension.gitHubRepositoryOwner

    val gitHubRepositoryName: String
        get() = extension.gitHubRepositoryName

    fun createGitHubClient(): GitHubClient {
        val client = GitHubClient()
        client.setSerializeNulls(false)
        client.setOAuth2Token(gitHubWriteToken)
        return client
    }
}
